extensions/kafka-eight-simpleConsumer/src/main/java/io/druid/firehose/kafka/KafkaEightSimpleConsumerFirehoseFactory.java

```diff
@@ -0,0 +1,305 @@
+          }
+        }
+        catch (InterruptedException e) {
+          //Let the caller decide whether to stop or continue when thread is interrupted.
+          log.warn(e, "Thread Interrupted while taking from queue, propagating the interrupt");
+          Thread.currentThread().interrupt();
+        }
+      }
+
+      @Override
+      public InputRow currRow()
+      {
+        if (stopped) {
+          return null;
+        }
+        return row;
+      }
+
+      @Override
+      public Committer makeCommitter()
+      {
+        final Map<Integer, Long> offsets = Maps.newHashMap(lastOffsetPartitions);
+
+        return new Committer()
+        {
+          @Override
+          public Object getMetadata()
+          {
+            return offsets;
+          }
+
+          @Override
+          public void run()
+          {
+
+          }
+        };
+      }
+
+      @Override
+      public void close() throws IOException
+      {
+        log.info("Stopping kafka 0.8 simple firehose");
+        stopped = true;
+        for (PartitionConsumerWorker t : consumerWorkers) {
+          t.close();
+        }
+      }
+    };
+  }
+
+  private static class PartitionConsumerWorker implements Closeable
+  {
+    private final String topic;
+    private final KafkaSimpleConsumer consumer;
+    private final int partitionId;
+    private final long startOffset;
+
+    private final AtomicBoolean stopped = new AtomicBoolean(false);
+    private volatile Thread thread = null;
+
+    PartitionConsumerWorker(String topic, KafkaSimpleConsumer consumer, int partitionId, long startOffset)
+    {
+      this.topic = topic;
+      this.consumer = consumer;
+      this.partitionId = partitionId;
+      this.startOffset = startOffset;
+    }
+
+    public void go(final LinkedBlockingQueue<BytesMessageWithOffset> messageQueue)
+    {
+      thread = new Thread()
+      {
+        @Override
+        public void run()
+        {
+          long offset = startOffset;
+          log.info("Start running parition[%s], offset[%s]", partitionId, offset);
+          try {
+            while (!stopped.get()) {
+              try {
+                Iterable<BytesMessageWithOffset> msgs = consumer.fetch(offset, CONSUMER_FETCH_TIMEOUT);
+                int count = 0;
+                for (BytesMessageWithOffset msgWithOffset : msgs) {
+                  offset = msgWithOffset.offset();
+                  messageQueue.put(msgWithOffset);
+                  count++;
+                }
+                log.debug("fetch [%s] msgs for partition [%s] in one time ", count, partitionId);
+              }
+              catch (InterruptedException e) {
+                log.info("Interrupted when fetching data, shutting down.");
+                return;
+              }
+              catch (Exception e) {
+                log.error(e, "Exception happened in fetching data, but will continue consuming");
```

Comment:
I assume this is simply to prevent firehose failure (rather than try to handle potentially transient errors), is that correct?